src/cortex_cli/cortex_cli.py

```diff
@@ -650,6 +659,73 @@ def _validate_cortex_cli_auth(no_auth, config_file) -> Optional[str]:
     return str(tokens_file)
 
 
+def _validate_measurements(input_circuit, results) -> Dict[str, List[str]]:
+    """Checks if the results of a circuit execution match the expected measurements in the input circuit."""
+    if results.measurements is None:
+        raise click.ClickException(f'No measurements obtained from backend. Job status is ${results.status}')
+
+    measured_qubits = {}
+    for instruction in input_circuit.instructions:
+        if instruction.name == 'measurement':
+            measured_qubits[instruction.args['key']] = list(instruction.qubits)
+
+    for measurement in results.measurements:
+        for measurement_name in measurement:
+            if measurement_name not in measured_qubits:
+                raise click.ClickException(
+                    'Measurements obtained from the backend \
+                     do not match measurements in the circuit.'
+                )
+
+    return measured_qubits
+
+
+def _human_readable_frequencies_output(input_circuit_name, measured_qubits, shots, results) -> str:
+    """Construct a human-readable output for the frequencies of the measured qubits."""
+    output_string = f'Circuit "{input_circuit_name}" results using '
+    output_string += f'calibration set {results.metadata.calibration_set_id} over {shots} shots:\n'
+
+    for measurement in results.measurements:
+        for measurement_name in measurement:
+            output_string += f'Frequencies of "{measurement_name}":\n'
+            output_string += '\t'.join(['state'] + measured_qubits[measurement_name]) + '\n'
+            shots_for_qubits = [
+                [shot[i] for shot in measurement[measurement_name]]
+                for i, qb in enumerate(measured_qubits[measurement_name])
+            ]
+            output_string += (
+                '\n'.join(
+                    [
+                        '\t'.join([str(value)] + [str(qb_shots.count(value) / shots) for qb_shots in shots_for_qubits])
+                        for value in [0, 1]
+                    ]
+                )
+                + '\n'
+            )
+    return output_string
+
+
+def _human_readable_shots_output(input_circuit_name, measured_qubits, shots, results) -> str:
```

Comment:
Maybe it would be nice to have multiple qubits next to each other:
shot qb1 qb2 qb3
0 0 1 0
1 1 0 0
2 0 0 1
...
I know we will hit the line limit at some point but than human readable format is probably not viable anyway.

Reply:
You mean across all measurements?